packages/teams-js/test/internal/validDomains.spec.ts

```diff
@@ -0,0 +1,369 @@
+import { GlobalVars } from '../../src/internal/globalVars';
+import { validateOrigin } from '../../src/internal/validDomains';
+import { app } from '../../src/public/app';
+import { _minRuntimeConfigToUninitialize } from '../../src/public/runtime';
+import { Utils } from '../utils';
+describe('validDomains', () => {
+  describe('testing main validDomain flow', () => {
+    let utils: Utils = new Utils();
+    beforeEach(() => {
+      // Set a mock window for testing
+      utils = new Utils();
+      utils.mockWindow.parent = undefined;
+      app._initialize(utils.mockWindow);
+      GlobalVars.isFramelessWindow = false;
+    });
+
+    afterAll(() => {
+      GlobalVars.isFramelessWindow = false;
+    });
+    afterEach(() => {
+      // Reset the object since it's a singleton
+      if (app._uninitialize) {
+        utils.setRuntimeConfig(_minRuntimeConfigToUninitialize);
+        app._uninitialize();
+      }
+    });
+    it('validateOrigin returns true if origin is in teams pre-known allowlist', async () => {
+      const messageOrigin = new URL('https://teams.microsoft.com');
+      const result = await validateOrigin(messageOrigin);
+      expect(result).toBe(true);
+    });
+    it('validateOrigin returns true if origin for subdomains in teams pre-known allowlist', async () => {
+      const messageOrigin = new URL('https://subdomain.teams.microsoft.com');
+      const result = await validateOrigin(messageOrigin);
+      expect(result).toBe(true);
+    });
+    it('validateOrigin returns false if origin is not in teams pre-known allowlist', async () => {
+      const messageOrigin = new URL('https://badorigin.com');
```

Comment:
>https://badorigin.com [](http://example.com/codeflow?start=37&length=21)

I know you are just copying this test over from the old location but in a future change consider using different domains for these tests. We shouldn't use domains in our code that we don't control. You could using something like badorigin.example.com instead.